src/main/java/org/candlepin/insights/task/queue/kafka/KafkaTaskQueueConfiguration.java

```diff
@@ -0,0 +1,138 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.insights.task.queue.kafka;
+
+import org.candlepin.insights.task.TaskFactory;
+import org.candlepin.insights.task.queue.TaskQueue;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.kafka.annotation.EnableKafka;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.config.KafkaListenerContainerFactory;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.listener.ConcurrentMessageListenerContainer;
+import org.springframework.kafka.support.serializer.ErrorHandlingDeserializer2;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+import org.springframework.kafka.support.serializer.JsonSerializer;
+
+import java.util.Map;
+
+
+/**
+ * A spring configuration that configures the required Beans to set up a KafkaTaskQueue.
+ *
+ * To enable this queue, set the following in the rhsm-conduit.properties file:
+ * <pre>
+ *     rhsm-conduit.tasks.queue=kafka
+ * </pre>
+ */
+@EnableKafka
+@Configuration
+@PropertySource("classpath:/rhsm-conduit.properties")
+public class KafkaTaskQueueConfiguration {
+
+    private static final String  TYPE_MAPPINGS = "task-message:" + TaskMessage.class.getCanonicalName();
+
+    @Autowired
+    private KafkaProperties kafkaProperties;
+
+    @Bean
+    @ConditionalOnProperty(prefix = "rhsm-conduit.tasks", name = "queue", havingValue = "kafka")
```

Comment:
The `ConditionalOnProperty` annotation can be applied to a class.  It looks like all the methods in this class have the same annotation, so maybe move it up to the class level?

Reply:
> The `ConditionalOnProperty` annotation can be applied to a class. It looks like all the methods in this class have the same annotation, so maybe move it up to the class level?

@awood I tried to do this initially but for some reason the configuration properties are not found when the the conditional annotation is checked. If the rhsm-conduit.tasks.queue property is set via an env var when the app is started all is good. It appears that the config file isn't loaded at that point.

I've tried various other ways to make this behave appropriately, but I'm coming up short. Anything stand out to you that I may be doing wrong, or suggestions on something else to try?

Reply:
Try adding a `@EnableConfigurationProperties(KafkaProperties.class)` annotation to the class and then setting the `@ConditionalOnProperty` annotation.  I think that will ensure the properties are loaded and populated.  If not, then don't worry about it. :)

Reply:
Same result. I think it has something to do with this:
https://github.com/spring-projects/spring-boot/issues/2282

Reply:
@awood  Despite my efforts I wasn't able to make this work. Pretty sure it is due to the issue sited above.